mig/shared/compat.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+#
+# --- BEGIN_HEADER ---
+#
+# support - helper functions for unit testing
+# Copyright (C) 2003-2024  The MiG Project by the Science HPC Center at UCPH
+#
+# This file is part of MiG.
+#
+# MiG is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# MiG is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# -- END_HEADER ---
+#
+
+"""This file contains an assortment of compatibility functions whose
+lifetime is intentionally intended to be time limited. We intentionally
+place these in their own namespace to make them easily identifiable and
+ease their subsequent removal."""
+
+from __future__ import absolute_import
+import codecs
+import sys
+
+PY2 = sys.version_info[0] < 3
+
+from mig.shared.base import STR_KIND, _force_default_coding
+
+
+def ensure_native_string(string_or_bytes):
+    """Given a supplied input which can be either a string or bytes
+    return a representation string operations while ensuring that its
```

Comment:
did a word or two fall out here? _providing_ string operations perhaps?

Reply:
Yep, looks like the diff captured some intermediate state of what I'd wanted that sentence to be. I'm almost certainly gonna be reading through if not this specifically then the compat module again soon so will make sure to go over this then.